ui/src/commonMain/kotlin/com/materiiapps/gloom/ui/utils/ColorUtils.kt

```diff
@@ -0,0 +1,30 @@
+package com.materiiapps.gloom.ui.utils
+
+import androidx.compose.ui.graphics.Color
+
+private val hexCodeRegex = "^#?(([0-9a-fA-F]{1,2})([0-9a-fA-F]{1,2})([0-9a-fA-F]{1,2}))$".toRegex()
+private val parsedColors = mutableMapOf<String, Color>()
+val String.parsedColor: Color
+    get() {
+        if (parsedColors.containsKey(this)) return parsedColors[this]!!
+        val match = hexCodeRegex.matchEntire(this) ?: return Color.Black
+
+        val (_, r, g, b) = match.groups.toList()
+            .mapNotNull { runCatching { it?.value?.toInt(16) }.getOrNull() }
+
+        return try {
+            parsedColors[this] = Color(r, g, b)
+            Color(r, g, b)
+        } catch (e: Throwable) {
+            Color.Black
+        }
+    }
+
+val Color.hexCode: String
+    inline get() {
+        val a: Int = (alpha * 255).toInt()
+        val r: Int = (red * 255).toInt()
+        val g: Int = (green * 255).toInt()
+        val b: Int = (blue * 255).toInt()
+        return "%02X%02X%02X%02X".format(r, g, b, a)
```

Comment:
format is pretty inefficient
you could do `.toInt().toString(16).padStart(2, '0')` for each component instead

Reply:
this isn't actually called very often, its only used to pass material3 colors to a webview, will be removed when i redo that